src/fu-plugin-list.c

```diff
@@ -198,6 +198,36 @@ fu_plugin_list_depsolve (FuPluginList *self, GError **error)
 			}
 		}
 
+		/* set priority as well */
+		for (guint i = 0; i < self->plugins->len; i++) {
+			FuPlugin *plugin = g_ptr_array_index (self->plugins, i);
+			deps = fu_plugin_get_rules (plugin, FU_PLUGIN_RULE_BETTER_THAN);
+			for (guint j = 0; j < deps->len && !changes; j++) {
+				const gchar *plugin_name = g_ptr_array_index (deps, j);
+				dep = fu_plugin_list_find_by_name (self, plugin_name, NULL);
+				if (dep == NULL) {
+					g_debug ("cannot find plugin '%s' "
```

Comment:
I think this should probably be a warning. It can easily happen if one plugin is compiled in and the other disabled.

Reply:
Is it a warning? If you have an optional shitty plugin then it might be that you want to mark the non-shitty version unconditionally better than the other one, without `ifdef`s.

Reply:
True, good point. I think you're right